posthog/api/error_tracking.py

```diff
@@ -97,27 +102,52 @@ def destroy(self, request, *args, **kwargs):
 
     def update(self, request, *args, **kwargs) -> Response:
         symbol_set = self.get_object()
-        symbol_set.delete()
         # TODO: delete file from s3
-        storage_ptr = upload_symbol_set(request.FILES["source_map"], self.team_id)
+        minified = request.FILES["minified"]
+        source_map = request.FILES["sourcemap"]
+        (storage_ptr, content_hash) = upload_symbol_set(minified, source_map, self.team_id)
         symbol_set.storage_ptr = storage_ptr
+        symbol_set.content_hash = content_hash
         symbol_set.save()
+        ErrorTrackingStackFrame.objects.filter(team=self.team, symbol_set=symbol_set).delete()
         return Response({"ok": True}, status=status.HTTP_204_NO_CONTENT)
 
 
-def upload_symbol_set(file, team_id) -> str:
+def upload_symbol_set(minified: UploadedFile, source_map: UploadedFile, team_id) -> tuple[str, str]:
+    js_data = construct_js_data_object(minified.read(), source_map.read())
+    content_hash = hashlib.sha512(js_data).hexdigest()
+
     try:
         if settings.OBJECT_STORAGE_ENABLED:
-            if file.size > FIFTY_MEGABYTES:
-                raise ValidationError(code="file_too_large", detail="Source maps must be less than 50MB")
+            # TODO - maybe a gigabyte is too much?
+            if len(js_data) > ONE_GIGABYTE:
+                raise ValidationError(
+                    code="file_too_large", detail="Combined source map and symbol set must be less than 1 gigabyte"
+                )
 
             upload_path = f"{settings.OBJECT_STORAGE_ERROR_TRACKING_SOURCE_MAPS_FOLDER}/{str(uuid7())}"
-            object_storage.write(upload_path, file)
-            return upload_path
+            object_storage.write(upload_path, bytes(js_data))
+            return (upload_path, content_hash)
         else:
             raise ObjectStorageUnavailable()
     except ObjectStorageUnavailable:
         raise ValidationError(
             code="object_storage_required",
             detail="Object storage must be available to allow source map uploads.",
         )
+
+
+def construct_js_data_object(minified: bytes, source_map: bytes) -> bytearray:
```

Comment:
Should we add a comment to this function and the equivalent in cymbal making sure to keep both implementations in sync?